config/schemas/config.yaml

```
@@ -14,6 +14,220 @@ $defs:
+                  operator: Exists
+    title: Kubernetes Affinity
+    type: object
+  component:
+    additionalProperties: false
+    description: This is meant to describe the base class if you will, for ck8s resources.
+    properties:
+      affinity:
+        $ref: '#/$defs/kubernetesAffinity'
+      enabled:
+        type: boolean
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      resources:
+        $ref: '#/$defs/resources'
+      tolerations:
+        $ref: '#/$defs/tolerations'
+      topologySpreadConstraints:
+        $ref: '#/$defs/topologySpreadConstraints'
+    required: [] # Shouldn't at least 'enabled' be required?
+    title: Common Resource
+    type: object
+  cpumem:
+    additionalProperties: false
+    properties:
+      cpu:
+        default: 100m
+        pattern: ^[1-9][0-9]*m?$
+        type:
+          - string
+          - integer
+      memory:
+        default: 128Mi
+        pattern: ^[0-9]+(\.[0-9]+)?([KMG]i)?$
+        type:
+          - string
+          - integer
+    title: CPU and Memory specifications
+    type: object
+  extraArgs:
+    description: Additional arguments for a thing???
+    items:
+      type: string
+    title: Extra arguments
+    type: array
+  fluentdBuffer:
+    additionalProperties:
+      $comment: Assuming that these are never structures, only scalars
+      not:
+        type:
+          - object
+          - array
+      type:
+        - string
+        - integer
+        - boolean
+    properties:
+      chunkLimitSize:
+        examples:
+          - 50MB
+        type: string
+      flushInterval:
+        examples:
+          - 15m
+        type: string
+      flushMode:
+        examples:
+          - interval
+        type: string
+      flushThreadBurstInterval:
+        type: number
+      flushThreadCount:
+        type: integer
+      retryForever:
+        type: boolean
+      retryMaxInterval:
+        type: integer
+      retryType:
+        examples:
+          - exponential_backoff
+        type: string
+      timekey:
+        examples:
+          - 10m
+        type: string
+      timekeyUseUtc:
+        type: boolean
+      timekeyWait:
+        examples:
+          - 1m
+        type: string
+      totalLimitSize:
+        examples:
+          - 9GB
+        type: string
+    type: object
+  goDuration:
+    examples:
+      - 2h45m0s
+    pattern: ^([-+]?[0-9]+(.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+    title: A more complexly expressed amount of time
+    type: string
+  iplist:
+    items:
+      oneOf:
+        - format: ipv4
+        - format: ipv6
+        - format: netmask
+      type: string
+    type: array
+  listen:
+    additionalProperties: false
+    properties:
+      ips:
+        $ref: '#/$defs/iplist'
+      ports:
+        $ref: '#/$defs/portlist'
+    type: object
+  nodeSelector:
+    additionalProperties:
+      type: string
+    examples:
+      - kubernetes.io/os: linux
+    type: object
+  pct:
```

Suggestion: Would it make sense to name this `percentage` instead of `pct` which might be a bit ambiguous IMO

```suggestion
  percentage:
```